external/fv3fit/fv3fit/train.py

```diff
@@ -112,6 +113,19 @@ def dump_dataclass(obj, yaml_filename):
             os.path.join(args.local_download_path, "validation")
         )
 
-    model = _get_model(train_config)
-    model.fit(train_batches)
+    model = fit_model(
```

Comment:
Would it be possible reduce the number of arguments on this `fit_model` function e.g. `fit_model(some_config, train_batches, val_batches)` and therefore make it easier to use. It seems like at minimum `model_type` and `hyperparamaters` could be grouped into a single object, and probably `input_variables` and `output_variables` too.

Reply:
I'm leaning a bit towards deleting `fit_model` entirely and putting its two lines here, after thinking about this.

If we enforce a one-to-one relationship, we can derive `model_type` from `type(hyperparameters)`. In the present API, many `model_type`s are allowed to share the same `hyperparameters` class (e.g. when we had aliases for the "random forest" model), which maybe doesn't make sense.

I excluded input/output variables because of the training function API, where I wanted to enforce input and output variables as clear required arguments. A flaw of the last design was grouping infrastructure-enforced, top-level arguments in the same class as model-specific hyperparameters, which led to model-specific hyperparameters getting used in top-level code. I think it's worthwhile limiting attributes on `hyperparameters` to model-specific details to avoid this in the future.

Reply:
OK, `fit_model` is inlined now which removes the `model_type` argument, and makes it clear that the training function and hyperparameters should both be derived from `model_type`.

Reply:
Inlining seems like a good solution.

> In the present API, many model_types are allowed to share the same hyperparameters class (e.g. when we had aliases for the "random forest" model), 

I hadn't realized this. This makes sense to me.